controllers/front/action.php

```diff
@@ -17,25 +18,366 @@
+    }
+
+    public function deleteWishlistAction($params)
+    {
+        if (isset($params['idWishlist'])) {
+            $wishlist = new wishlist($params['idWishlist']);
+
+            if (true === $wishlist->delete()) {
+                $this->ajaxDie(
+                    Tools::jsonEncode([
+                        'success' => true,
+                        'message' => $this->module->l('Wishlist has been removed', 'mywishlist')
+                    ])
+                );
+            }
+
+            $this->ajaxDie(
+                Tools::jsonEncode([
+                    'success' => false,
+                    'message' => $this->module->l("Wishlist couldn't been removed", 'mywishlist')
+                ])
+            );
+        }
+
+        $this->ajaxDieMissingParams();
     }
 
-    public function AddToAWishlistAction($params)
+    public function deleteProductFromWishlistAction($params)
     {
-        dump($params);
+        if (
+            isset($params['idWishlist'])
+            && isset($params['id_product'])
+            && isset($params['id_product_attribute'])
+            && isset($params['id_customer'])
+        ) {
+            $isDeleted = wishlist::removeProduct(
+                $params['idWishlist'],
+                $params['id_customer'],
+                $params['id_product'],
+                $params['id_product_attribute']
+            );
+
+            if (true === $isDeleted) {
+                $this->ajaxDie(
+                    Tools::jsonEncode([
+                        'success' => true,
+                        'message' => $this->module->l('Product succesfully removed', 'mywishlist')
+                    ])
+                );
+            }
+
+            $this->ajaxDie(
+                Tools::jsonEncode([
+                    'success' => false,
+                    'message' => $this->module->l('Unable to remove product from wishlist', 'mywishlist')
+                ])
+            );
+        }
+
+        $this->ajaxDieMissingParams();
+    }
+
+    public function updateProductFromWishlistAction($params)
+    {
+        if (isset(
+            $params['idWishlist'],
+            $params['id_product'],
+            $params['id_product_attribute'],
+            $params['priority'],
+            $params['quantity']
+        )) {
+            $isDeleted = wishlist::updateProduct(
+                $params['idWishlist'],
+                $params['id_product'],
+                $params['id_product_attribute'],
+                $params['priority'],
+                $params['quantity']
+            );
+
+            if ($isDeleted) {
+                $this->ajaxDie(
+                    Tools::jsonEncode([
+                        'success' => true,
+                        'message' => $this->module->l('Product succesfully updated', 'mywishlist')
+                    ])
+                );
+            }
+
+            $this->ajaxDie(
+                Tools::jsonEncode([
```

Comment:
```suggestion
                json_encode([
```
https://github.com/PrestaShop/PrestaShop/blob/1.7.6.x/classes/Tools.php#L3007